query-connector/e2e/customize_query.spec.ts

```diff
@@ -0,0 +1,193 @@
+    await expect(
+      page
+        .getByRole("table")
+        .filter({ hasText: "doxycycline hyclate 100 MG" })
+        .getByRole("row"),
+    ).toHaveCount(2);
+  });
+
+  test("customize query select / deselect all filters whole DibbsConceptType, across tabs", async ({
+    page,
+  }) => {
+    // await expect(
+    //   page.getByText("249 labs found, 4 medications found, 104 conditions found."),
+    // ).toBeVisible();
+    await page.getByRole("link", { name: "Labs" }).click();
+    await page.getByRole("button", { name: "Deselect all labs" }).click();
+
+    // Spot check a couple valuesets for deselection
+    await expect(page.getByText("0 of 38 selected")).toBeVisible();
+    await expect(page.getByText("0 of 14 selected")).toBeVisible();
+    await expect(page.getByText("0 of 33 selected")).toBeVisible();
+
+    // Now de-select all the medications via the group check marks
+    await page.getByRole("link", { name: "Medications" }).click();
+    await page
+      .getByRole("button", { name: "Chlamydia Medication" })
+      .getByRole("img")
+      .click();
+    await expect(page.getByText("0 of 4 selected")).toBeVisible();
+
+    await page.getByRole("button", { name: "Apply changes" }).click();
+    await expect(
+      page.getByRole("alert").getByText("Query Customization Successful!"),
+    ).toBeVisible();
+
+    await page.getByRole("button", { name: "Submit" }).click();
+    await expect(page.getByText("Loading")).toHaveCount(0, { timeout: 10000 });
+
+    // Make sure we have a results page with a single patient
+    // Non-interactive 'div' elements in the table should be located by text
+    await expect(
+      page.getByRole("heading", { name: "Patient Record" }),
+    ).toBeVisible();
+    await expect(page.getByText("Patient Name")).toBeVisible();
+    await expect(page.getByText(TEST_PATIENT_NAME)).toBeVisible();
+    await expect(page.getByText("Patient Identifiers")).toBeVisible();
+    await expect(
+      page.getByText(`Medical Record Number: ${TEST_PATIENT.MRN}`),
+    ).toBeVisible();
+
+    // Should be no medication requests available
+    await expect(
+      page.getByRole("button", { name: "Medication Requests" }),
+    ).not.toBeVisible();
+
+    // Eliminating all value set labs should also remove diagnostic reports
+    await expect(
+      page.getByRole("button", { name: "Diagnostic Reports" }),
+    ).not.toBeVisible();
+
+    // Observations table should have 5 rows, all of which are SDoH factors rather than lab results
+    await expect(
+      page.getByRole("button", { name: "Observations", expanded: true }),
+    ).toBeVisible();
+    await expect(
+      page
+        .getByRole("table")
+        .filter({ hasText: "I do not have housing" })
+        .getByRole("row"),
+    ).toHaveCount(6);
+    const acceptableSdohKeywords = [
+      "history",
+      "narrative",
+      "housing",
+      "pregnancy",
+      "with anonymous partner",
```

Comment:
This might be out of scope for the ticket (or honestly over optimization at this stage), but what do you think about creating a customized query constant like we do for `TEST_PATIENT`? We could add key words like this as well as the expected medications, etc. I'm not sure if it's easier to update a constant or just update the tests at this point but thought I'd throw it out there.